src/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { createContext, useContext, useState, useEffect, useCallback, ReactNode } from "react";
+import { Whoami, getOpenIdManager } from "./openid-manager";
+
+// Placeholder types for auth state and actions
+interface AuthUser {
+  id: string;
+  email: string;
+  name?: string;
+}
+
+interface AuthState {
+  isAuthenticated: boolean;
+  user: AuthUser | null;
+  token: string | null;
+  isLoading: boolean;
+  error: string | null;
+}
+
+interface AuthContextType extends AuthState {
+  signIn: () => Promise<void>;
+  signOut: () => Promise<void>;
+  refreshToken: () => Promise<void>;
+  register: () => Promise<void>;
+  handleRedirect: (url: URL) => Promise<void>;
+}
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+const openidManager = getOpenIdManager();
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [state, setState] = useState<AuthState>({
+    isAuthenticated: false,
+    user: null,
+    token: null,
+    isLoading: true,
+    error: null,
+  });
+
+  // On mount, check for existing session and fetch user info
+  useEffect(() => {
```

Comment:
This useEffect is untested AI slop. Need the token to come through before doing something with it